airflow/utils/log/file_task_handler.py

```diff
@@ -404,11 +406,15 @@ def _read(
         )
         log_pos = len(logs)
         messages = "".join([f"*** {x}\n" for x in messages_list])
+        end_of_log = ti.try_number != try_number or ti.state not in (
```

Comment:
Nitpick: Since you check whether `ti.state` is or isn't in `(TaskInstanceState.RUNNING, TaskInstanceState.DEFERRED)` twice, would it make sense to replace `is_running` with `is_in_running_or_deferred` and use that when checking for `executor_messages`/`remote_logs` and when defining `end_of_log`?